### ISSUE TITLE
Provide more context for "Image not used in any deploy config"

### ISSUE THREAD

**Issue by jazzdan** [tilt-dev/tilt#3410]
A common error that people encounter when using Tilt is "Image not used in any deploy config". It looks like this:

```
Image not used in any deploy config:
    ✕ 727006795293.ecr.us-east-1.amazonaws.com/foo
```
This can be really confusing because if you have a complicated set up it won't be obvious where `727006795293.ecr.us-east-1.amazonaws.com/foo` is coming. It might be being generated in your Tiltfile via a `local` and could also be interacting with [`default_registry`](https://docs.tilt.dev/personal_registry.html). In addition it won't be obvious where to look for image in your YAML. It's likely that the YAML is being templated if not outright generated. Is the image name an environment variable, computed somewhere else, in the Helm chart?

I think it would help if we listed the image namees that Tilt knows about alongside this error message? So like:

```
Image not used in any deploy config:
    ✕ 727006795293.ecr.us-east-1.amazonaws.com/foo
Images that Tilt has seen in all deploy configs:
    *  foo
```

But there are probably other/better ways to solve this too.

cc @jgoulah

**Comment by nicks** [comment]
For reference, this is what the error message currently looks like:

```
Image not used in any deploy config:
    ✕ gcr.typo.io/foo
Did you mean…
    - gcr.io/foo
Skipping this image build
```

A common problem I've seen is that the user doesn't know what a "Deploy config" is, and doesn't know that they're supposed to be providing k8s configs. is that what's happening here?

**Comment by jazzdan** [comment]
@nicks good point, but that's only true if we find any similar matches. I think this is a case where a misconfiguration is a causing `k8s_yaml` for `gcr.io/foo` to not get called at all.

**Comment by jgoulah** [comment]
in this case it was that our yaml being passed to k8s_yaml() wasn't being captured correctly due to a bug in our stack, I think any additional output would be great though as this error has popped up for various reasons

**Comment by victorwuky** [comment]
@jgoulah : This is now available in the latest Tilt release: https://github.com/tilt-dev/tilt/releases/tag/v0.14.3
